gio-macros/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+// Take a look at the license at the top of the repository in the LICENSE file.
+
+extern crate proc_macro;
+
+mod action_map;
+
+use proc_macro::TokenStream;
+use syn::{parse_macro_input, ItemImpl};
+
+/// Macro for creating [`gio::Action`]s and registering them in a given [`gio::ActionMap`].
+/// It generates a method `register_actions` with a following signature:
+///
+/// ```rust
+/// # struct X; impl X {
+/// fn register_actions<AM: glib::object::IsA<gio::ActionMap>>(&self, map: &AM)
+/// # {} }
+/// ```
+///
+/// # Example
+/// ```rust
+/// use gio::prelude::*;
+///
+/// #[derive(glib::Downgrade)]
+/// pub struct MyApplication(gio::Application);
+///
+/// impl MyApplication {
+///     pub fn new() -> Self {
+///         let app = Self(gio::Application::new(
+///             Some("com.example.MyApplication"),
+///             gio::ApplicationFlags::FLAGS_NONE,
+///         ));
+///         app.register_actions(&app.0);
+///         app
+///     }
+/// }
+///
+/// #[gio::actions]
+/// impl MyApplication {
+///     fn action1(&self) {
+///         // handle "action1"
+///     }
+///
+///     // Explicitely specify action name
+///     #[action(name = "action_second")]
+///     fn action2(&self) {
+///         // handle "action_second"
+///     }
+///
+///     // Action with a parameter
```

Comment:
Does it also handle return values?

Reply:
There are stateful actions though. `Action::change_state()` / `Action::get_state()`

Reply:
This macro doesn't cover/implement stateful actions.

Reply:
Please document that then :)